File: src/integrationTest/java/uk/gov/hmcts/reform/bulkscan/orchestrator/util/PortWaiter.java
package uk.gov.hmcts.reform.bulkscan.orchestrator.util;

import java.io.Closeable;
import java.io.IOException;
import java.net.Socket;

import static java.util.concurrent.TimeUnit.SECONDS;
import static org.awaitility.Awaitility.await;

public final class PortWaiter {

    public static void waitFor(int port, long timeoutSeconds) {
        await()
            .atMost(timeoutSeconds, SECONDS)
            .ignoreExceptions()
            .until(() -> {
                try (Closeable socket = new Socket("localhost", port)) {
                    return true;
                } catch (IOException exception) {
                    return false;
                }
            });
    }

    public static void waitFor(int port) {
        waitFor(port, 5L);
    }

    private PortWaiter() {
        // utility class constructor
    }
}

File: src/integrationTest/java/uk/gov/hmcts/reform/bulkscan/orchestrator/controllers/config/PortWaiter.kt
package uk.gov.hmcts.reform.bulkscan.orchestrator.controllers.config

import org.awaitility.Awaitility.await
import java.net.Socket
import java.util.concurrent.TimeUnit.SECONDS

object PortWaiter {
    fun waitFor(port: Int, timeoutSeconds: Long = 5L) {
        await()
            .atMost(timeoutSeconds, SECONDS)
            .ignoreExceptions()
            .until { Socket("localhost", port).use { true } }
    }
}
